pkg/providers/amifamily/al2023.go

```diff
@@ -0,0 +1,81 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"fmt"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	corev1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/amifamily/bootstrap"
+)
+
+type AL2023 struct {
+	DefaultFamily
+	*Options
+}
+
+func (a AL2023) DefaultAMIs(version string) []DefaultAMIOutput {
+	return []DefaultAMIOutput{
+		{
```

Comment:
Not at launch, I don't believe I can share any timeline details here.

Reply:
It seems like we should still _allow_ users to launch instances that have Nvidia GPUs and just install the drivers themselves, at least until we formally support a version of the image that has it out-of-the-box.

Reply:
If I'm reading it correctly, we are saying that the architecture of the instance type that we are using must be amd64 and also that it can't contain a GPU in the GPU count section, which inherently means that this image version is incompatible with a GPU instance type. I think that we need to drop this requirement from both of the default images here.

Reply:
You're right, I was reviewing on my phone and going off memory. The GPU not exists requirements were originally not there but I had copied the AL2 SSM parameters over briefly and they stayed when I reverted. Their gone now, only requirements are architecture now.